keras_nlp/models/preprocessor.py

```diff
@@ -147,9 +161,44 @@ def from_preset(
                     f"Found multiple possible subclasses {names}. "
                     "Please call `from_preset` on a subclass directly."
                 )
-            cls = subclasses[0]
-        tokenizer = load_from_preset(
+
+        if check_file_exists(preset, PREPROCESSOR_CONFIG_FILE):
+            get_file(preset, PREPROCESSOR_CONFIG_FILE)
+            preprocessor = load_serialized_object(
+                preset,
+                PREPROCESSOR_CONFIG_FILE,
+            )
+            for asset in preprocessor.tokenizer.file_assets:
+                get_file(preset, os.path.join(TOKENIZER_ASSET_DIR, asset))
+            tokenizer_asset_dir = get_asset_dir(
+                preset,
+                config_file=TOKENIZER_CONFIG_FILE,
+                asset_dir=TOKENIZER_ASSET_DIR,
+            )
+            preprocessor.tokenizer.load_assets(tokenizer_asset_dir)
```

Comment:
we could early return here if the branching is getting too much, but either way.